tests/unit/lms/views/api/canvas/files_test.py

```diff
@@ -1,113 +1,40 @@
 import pytest
```

Comment:
These tests were a bit weird and had quite a few tests to show things _didn't_ happen. Not sure about that as the list of things this code should _not_ do is infinite. In any case, they are much smaller and appropriate to the size of the view now.

Reply:
Hmm. I think `test_it_doesnt_catch_CanvasAPIErrors_from_list_files()` might have better been named `test_it_raises_CanvasAPIError_if_list_files_raises()`. It's not really a test for what `FilesAPIView` _doesn't_ do. It's a test for something it positively _does_ do. `FilesAPIView.list_files` should have a `:raise CanvasAPIError: if something goes wrong with the Canvas API request(s)` on it, and this test is effectively a test for that part of the docstring.

_But_ I think that putting `:raise`'s on docstrings is more something for services and other things that get called by our own code. The point is that if I'm writing a new line of code that calls `FooService.foo()` I can look at the docstring and see if there are any raises I need to be aware of. That applies less to views which we don't call ourselves.

I can see some value in it: if we have an exception view for a given exception, `:raise` docstrings could help with grepping for all views that might trigger that exception view. But I wouldn't rate our chances of successfully adding all the right `:raise`'s to all the right views. :man_shrugging:

Reply:
I agree about the multiple assertions per unit test

Reply:
I can see the point of `test_it_doesnt_check_whether_the_file_is_in_the_course()` too since we have specific reasons why we want to be sure that student launches don't trigger this check

Reply:
I've parameterised the test to pass in both values. This ensures we don't always pass a fixed value, but there's no other behavior at all (we just pass it through) so I think that aspect is covered.

I disagree with the suggested names. This code does not raise any exceptions, so `test_it_raises_CanvasAPIError_if_list_files_raises` is inaccurate unless we were to catch and re-raise. For this reason `test_it_doesnt_catch_CanvasAPIErrors_from_list_files` is a more accurate name. Also phrasing this positively doesn't change the scenario under test, which is testing that we didn't implement any code to catch this. So this is testing a thing we didn't implement, rather than something we did.

The problem withthis is we also don't catch `ValueError`, or `OSError` or any thing form the helper function, or modify the user, or delete the account, or... There's kind of no end to it.

I'm similarly uncomfortable with documenting errors raised by code we call, yet don't raise directly ourselves for a number of reasons:

 * Is this even helpful? Like have you actually read this and found it to be helpful? I could imagine it would mostly make searching for a name more difficult
 * Where do you stop? One layer down, two, all the way until you hit C?
 * If something at the bottom of the stack raises a new error, do we really want coupled documentation that requires us to document that in every class above?
 * Do we want to repeatedly document the reasons for an exception further and further from the code in charge? Will these be accurate over time.
 
I have more sympathy for this in a few cases:

 * The errors are raised by private functions of code and exposed, but not raised directly, in a public interface
 * Service type boundaries as you mention